### PR TITLE
Added fuzzing infrastructure for OSS-Fuzz. Fuzzer similar to cargo-fu…

### PR DESCRIPTION
**Pull request by DavidKorczynski**
…zz targets.

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
Cr https://github.com/rust-lang/regex/issues/650

File: fuzz/fuzz_targets/fuzz_regex_match.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#![no_main]
+use libfuzzer_sys::fuzz_target;
+
+fuzz_target!(|data: &[u8]| {
+    if let Ok(data) = std::str::from_utf8(data) {
+        // split data into regular expression and actual input to search through
+        use std::cmp::max;
+        let len = data.chars().count();
+        let split_off_point = max(len / 5, 1) as usize;
```
Comment (ethanpailes) [inline fuzz/fuzz_targets/fuzz_regex_match.rs:9]: What if this split point was determined by the first byte or two of the data so the fuzzer the regex and input size were not locked to one another?

Comment (DavidKorczynski) [reply fuzz/fuzz_targets/fuzz_regex_match.rs]: Good idea - how does the new one look?

**Comment by DavidKorczynski** [comment]
Was wondering if we could land this @ethanpailes, @BurntSushi - all seems to be ready from OSS-Fuzz side of things.

File: fuzz/fuzz_targets/fuzz_regex_match.rs (hunk 1 above)
Comment (ethanpailes) [reply fuzz/fuzz_targets/fuzz_regex_match.rs]: looks good to me, but it's only @BurntSushi's opinion that matters here.

**Comment by DavidKorczynski** [comment]
ping - any updates here @BurntSushi ?

**Comment by BurntSushi** [comment]
@DavidKorczynski Sorry for dragging my feet on this. I merged this PR in #718. What are the next steps?

**Pull request closed without merging**